### PR TITLE
DRILL-6961: Handle exceptions during queries to information_schema

### PR DESCRIPTION
**Pull request by agozhiy**
https://issues.apache.org/jira/browse/DRILL-6961

### CODE REVIEW

File: contrib/storage-kafka/src/main/java/org/apache/drill/exec/store/kafka/schema/KafkaMessageSchema.java [hunk 1]
```diff
@@ -76,11 +75,10 @@ public Table getTable(String tableName) {
...
         tableNames = kafkaConsumer.listTopics().keySet();
-      } catch(KafkaException e) {
-        throw UserException.dataReadError(e).message("Failed to get tables information").addContext(e.getMessage())
-            .build(logger);
+      } catch (Exception e) {
+        logger.warn("Failed to get tables information: {}", e.getMessage());
       }
     }
-    return tableNames;
+    return tableNames != null ? tableNames : Collections.emptySet();
```
Comment (vvysotskyi) [inline contrib/storage-kafka/src/main/java/org/apache/drill/exec/store/kafka/schema/KafkaMessageSchema.java]: `tableNames` can be `null` only for the case when the exception was caught. Is it makes sense to add return statement there and revert the change in this line?

File: contrib/storage-opentsdb/src/main/java/org/apache/drill/exec/store/openTSDB/client/services/ServiceImpl.java [hunk 2]
```diff
@@ -101,10 +102,9 @@ public ServiceImpl(String connectionURL) {
     try {
       return client.getAllTablesName().execute().body();
     } catch (IOException e) {
-      throw UserException.connectionError(e)
-              .message("Cannot connect to the db. " +
-                      "Maybe you have incorrect connection params or db unavailable now")
-              .build(log);
+      log.warn("Cannot connect to the db. " +
```
Comment (vvysotskyi) [inline contrib/storage-opentsdb/src/main/java/org/apache/drill/exec/store/openTSDB/client/services/ServiceImpl.java]: Please also pass `e.getCause()`.

File: contrib/storage-opentsdb/src/test/java/org/apache/drill/store/openTSDB/TestOpenTSDBPlugin.java [hunk 3]
```diff
@@ -185,4 +190,46 @@ public void testDescribe() throws Exception {
...
+    storagePluginConfig.setEnabled(true);
+    pluginRegistry.createOrUpdate(OpenTSDBStoragePluginConfig.NAME, storagePluginConfig, true);
+    test("select * from information_schema.`views`", cluster);
+  }
+
+  private void test(String query) throws Exception {
+    test(query, cluster);
+  }
+
+  private void test(String query, ClusterFixture cluster) throws Exception {
```
Comment (vvysotskyi) [inline contrib/storage-opentsdb/src/test/java/org/apache/drill/store/openTSDB/TestOpenTSDBPlugin.java]: In most of the cases where this method is used, `cluster` from the parent class is used, and its usage may be replaced with `queryBuilder()`, so please remove this method and add corresponding changes to simplify tests.

File: contrib/storage-opentsdb/src/test/java/org/apache/drill/store/openTSDB/TestOpenTSDBPlugin.java [hunk 4]
```diff
@@ -185,4 +190,46 @@ public void testDescribe() throws Exception {
...
+
+  private String getPlan(String query) throws Exception {
+    return cluster.clientFixture()
+        .queryBuilder()
+        .sql(query)
+        .explainJson();
+  }
+
+  private void testPhysicalPlanExecution(String plan) throws Exception {
+    cluster.clientFixture()
```
Comment (vvysotskyi) [inline contrib/storage-opentsdb/src/test/java/org/apache/drill/store/openTSDB/TestOpenTSDBPlugin.java]: Please replace 
```
    cluster.clientFixture()
        .queryBuilder()
```
with `queryBuilder()` call.

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/StoragePluginRegistryImpl.java [hunk 5]
```diff
@@ -581,7 +581,11 @@ public void registerSchemas(SchemaConfig schemaConfig, SchemaPlus parent) throws
 
         // finally register schemas with the refreshed plugins
         for (StoragePlugin plugin : enabledPlugins.plugins()) {
-          plugin.registerSchemas(schemaConfig, parent);
+          try {
+            plugin.registerSchemas(schemaConfig, parent);
+          } catch (Exception e) {
+            logger.warn("Error during `{}` schema initialization: {}", plugin.getName(), e.getMessage());
```
Comment (vvysotskyi) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/StoragePluginRegistryImpl.java]: `e.getCause()`

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/ischema/InfoSchemaRecordGenerator.java [hunk 6]
```diff
@@ -379,7 +379,7 @@ public Views(OptionManager optionManager) {
     public boolean visitTable(String schemaName, String tableName, Table table) {
       if (table.getJdbcTableType() == TableType.VIEW) {
         records.add(new Records.View(IS_CATALOG_NAME, schemaName, tableName,
-                    ((DrillViewInfoProvider) table).getViewSql()));
+            table instanceof DrillViewInfoProvider ? ((DrillViewInfoProvider) table).getViewSql() : ""));
```
Comment (vvysotskyi) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/ischema/InfoSchemaRecordGenerator.java]: Could you please add a comment that view SQL is not available for the case of some non-Drill views, for example, JDBC view.

File: contrib/storage-opentsdb/src/test/java/org/apache/drill/store/openTSDB/TestOpenTSDBPlugin.java [hunk 7]
```diff
@@ -185,4 +190,46 @@ public void testDescribe() throws Exception {
...
+
+  private void test(String query, ClusterFixture cluster) throws Exception {
+    cluster.clientFixture().runQueriesAndLog(query);
+  }
+
+  private int testSql(String query) throws Exception {
+    return (int) cluster.clientFixture()
+        .queryBuilder()
+        .sql(query)
+        .print();
```
[the remark this answers is not in the document]
Comment (agozhiy) [reply contrib/storage-opentsdb/src/test/java/org/apache/drill/store/openTSDB/TestOpenTSDBPlugin.java]: Done.

File: contrib/storage-opentsdb/src/test/java/org/apache/drill/store/openTSDB/TestOpenTSDBPlugin.java [hunk 8]
```diff
@@ -185,4 +190,46 @@ public void testDescribe() throws Exception {
...
+    pluginRegistry.createOrUpdate(OpenTSDBStoragePluginConfig.NAME, storagePluginConfig, true);
+    test("select * from information_schema.`views`", cluster);
+  }
+
+  private void test(String query) throws Exception {
+    test(query, cluster);
+  }
+
+  private void test(String query, ClusterFixture cluster) throws Exception {
+    cluster.clientFixture().runQueriesAndLog(query);
```
[the remark this answers is not in the document]
Comment (agozhiy) [reply contrib/storage-opentsdb/src/test/java/org/apache/drill/store/openTSDB/TestOpenTSDBPlugin.java]: Done.

File: contrib/storage-opentsdb/src/test/java/org/apache/drill/store/openTSDB/TestOpenTSDBPlugin.java [hunk 9]
```diff
@@ -185,4 +190,46 @@ public void testDescribe() throws Exception {
...
+  }
+
+  private String getPlan(String query) throws Exception {
+    return cluster.clientFixture()
+        .queryBuilder()
+        .sql(query)
+        .explainJson();
+  }
+
+  private void testPhysicalPlanExecution(String plan) throws Exception {
```
[the remark this answers is not in the document]
Comment (agozhiy) [reply contrib/storage-opentsdb/src/test/java/org/apache/drill/store/openTSDB/TestOpenTSDBPlugin.java]: Done.

File: contrib/storage-opentsdb/src/test/java/org/apache/drill/store/openTSDB/TestOpenTSDBPlugin.java [hunk 10]
```diff
@@ -185,4 +190,46 @@ public void testDescribe() throws Exception {
...
+  }
+
+  private int testSql(String query) throws Exception {
+    return (int) cluster.clientFixture()
+        .queryBuilder()
+        .sql(query)
+        .print();
+  }
+
+  private String getPlan(String query) throws Exception {
```
[the remark this answers is not in the document]
Comment (agozhiy) [reply contrib/storage-opentsdb/src/test/java/org/apache/drill/store/openTSDB/TestOpenTSDBPlugin.java]: Done.

File: contrib/storage-opentsdb/src/test/java/org/apache/drill/store/openTSDB/TestOpenTSDBPlugin.java [hunk 11]
```diff
@@ -185,4 +190,46 @@ public void testDescribe() throws Exception {
...
+  private void test(String query) throws Exception {
+    test(query, cluster);
+  }
+
+  private void test(String query, ClusterFixture cluster) throws Exception {
+    cluster.clientFixture().runQueriesAndLog(query);
+  }
+
+  private int testSql(String query) throws Exception {
+    return (int) cluster.clientFixture()
```
[the remark this answers is not in the document]
Comment (agozhiy) [reply contrib/storage-opentsdb/src/test/java/org/apache/drill/store/openTSDB/TestOpenTSDBPlugin.java]: Done.

File: contrib/storage-opentsdb/src/main/java/org/apache/drill/exec/store/openTSDB/client/services/ServiceImpl.java (hunk 2 above)
Comment (agozhiy) [reply contrib/storage-opentsdb/src/main/java/org/apache/drill/exec/store/openTSDB/client/services/ServiceImpl.java]: Done.

File: contrib/storage-opentsdb/src/test/java/org/apache/drill/store/openTSDB/TestOpenTSDBPlugin.java (hunk 3 above)
Comment (agozhiy) [reply contrib/storage-opentsdb/src/test/java/org/apache/drill/store/openTSDB/TestOpenTSDBPlugin.java]: Done.

File: contrib/storage-opentsdb/src/test/java/org/apache/drill/store/openTSDB/TestOpenTSDBPlugin.java (hunk 4 above)
Comment (agozhiy) [reply contrib/storage-opentsdb/src/test/java/org/apache/drill/store/openTSDB/TestOpenTSDBPlugin.java]: Done.

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/StoragePluginRegistryImpl.java (hunk 5 above)
Comment (agozhiy) [reply exec/java-exec/src/main/java/org/apache/drill/exec/store/StoragePluginRegistryImpl.java]: Done.

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/ischema/InfoSchemaRecordGenerator.java (hunk 6 above)
Comment (agozhiy) [reply exec/java-exec/src/main/java/org/apache/drill/exec/store/ischema/InfoSchemaRecordGenerator.java]: Done.

**Comment by agozhiy** [comment]
Done.
